python/pyarrow/tests/test_fs.py

```diff
@@ -1321,22 +1321,24 @@ def test_s3fs_wrong_region():
     from pyarrow.fs import S3FileSystem
 
     # wrong region for bucket
-    fs = S3FileSystem(region='eu-north-1')
+    # anonymous=True incase CI/etc has invalid credentials
+    fs = S3FileSystem(region='eu-north-1', anonymous=True)
 
     msg = ("When getting information for bucket 'voltrondata-labs-datasets': "
            r"AWS Error UNKNOWN \(HTTP status 301\) during HeadBucket "
            "operation: No response body. Looks like the configured region is "
-           "'eu-north-1' while the bucket is located in 'us-east-2'."
-           "|NETWORK_CONNECTION")
-    with pytest.raises(OSError, match=msg) as exc:
+           "'eu-north-1' while the bucket is located in 'us-east-2'.")
+    with pytest.raises(OSError) as exc:
         fs.get_file_info("voltrondata-labs-datasets")
 
-    # Sometimes fails on unrelated network error, so next call would also fail.
-    if 'NETWORK_CONNECTION' in str(exc.value):
+    # Sometimes fails on unrelated network or permission error, so next
+    # call would also fail.
+    if 'UNKNOWN' not in str(exc.value):
```

Comment:
Why "not in"? Don't we want this condition to be a bit more specific?

Reply:
Ya, good point. Since the main issue was `anonymous=True` needing to be set, I'll reset everything else.